rootfs/deploy.py

```diff
@@ -36,6 +36,40 @@ def log(msg):
         print(msg)
 
 
+def get_registry_name():
+    hostname = os.getenv('DEIS_REGISTRY_HOSTNAME', "").replace("https://", "").replace("http://", "")
+    if regsitryLocation == "off-cluster":
+        organization = os.getenv('DEIS_REGISTRY_ORGANIZATION')
+        regName = ""
+        # empty hostname means dockerhub and hence no need to prefix the image
+        if hostname != "":
+            regName = hostname + "/"
+        # Registries may have organizations/namespaces under them which needs to
+        # be prefixed to the image
+        if organization != "":
+            regName = regName + organization
+        return regName
+    elif regsitryLocation == "ecr":
+        return hostname
+    elif regsitryLocation == "gcr":
+        return hostname+"/"+os.getenv('DEIS_REGISTRY_GCS_PROJ_ID')
+    else:
+        return os.getenv("DEIS_REGISTRY_SERVICE_HOST") + ":" + os.getenv("DEIS_REGISTRY_SERVICE_PORT")
+
+
+def docker_push(client, repo, tag):
+    if regsitryLocation != "on-cluster":
+        hostname = os.getenv('DEIS_REGISTRY_HOSTNAME', 'https://index.docker.io/v1/')
+        auth_config = {
+            'username': os.getenv('DEIS_REGISTRY_USERNAME'),
+            'password': os.getenv('DEIS_REGISTRY_PASSWORD'),
+            'serveraddress': hostname,
+        }
+        return client.push(repo, tag=tag, stream=True, auth_config=auth_config)
+    else:
+        return client.push(repo, tag=tag, stream=True, insecure_registry=True)
```

Comment:
Ah - I did a PR that pulls out some of `insecure_registry` stuff but not fixing the env vars in Controller https://github.com/deis/controller/pull/933